File: src/main/java/com/jcloisterzone/event/FlierDiceRollEvent.java
package com.jcloisterzone.event;

import com.jcloisterzone.board.Position;
import io.vavr.collection.Set;

public class FlierDiceRollEvent extends PlayEvent {

    private Set<Position> positions;
    private final int value;
    private final String type;

    public FlierDiceRollEvent(PlayEventMeta metadata, Set<Position> positions, int value, String type) {
        super(metadata);
        this.positions = positions;
        this.value = value;
        this.type = type;
    }

    public int getValue() {
        return value;
    }

    public String getType() {
        return type;
    }

    public Set<Position> getPositions() {
        return positions;
    }
}

File: src/main/ts/com/jcloisterzone/event/FlierDiceRollEvent.ts
import type { Position } from "../board/Position.js";
import type { Set } from "../../../io/vavr/Set.js";
import { PlayEvent, type PlayEventMeta } from "./PlayEvent.js";

/** A meteorite d3-style roll (standard/extended variants): the impacted `positions`, the
 *  derived `value`, and a `type` discriminator (`meteorite-impact.<variant>`). */
export class FlierDiceRollEvent extends PlayEvent {
  static readonly simpleName = "FlierDiceRollEvent";

  constructor(
    metadata: PlayEventMeta,
    private readonly positions: Set<Position>,
    private readonly value: number,
    private readonly type: string,
  ) {
    super(metadata);
  }

  getPositions(): Set<Position> {
    return this.positions;
  }
  getValue(): number {
    return this.value;
  }
  getType(): string {
    return this.type;
  }
}
